Objective-C/CBLQuery.mm

```diff
@@ -63,6 +68,12 @@ - (instancetype) initWithDatabase: (CBLDatabase*)database
         _database = database;
         _json = json;
         _language = kC4JSONQuery;
+        
+        // Return error if the query is not compiled;
+        NSError* error = nil;
+        if (![self compile: &error])
+            [NSException raise: NSInvalidArgumentException
```

Comment:
Can you also log the error for the case that somebody experiences a crash due to this so they have some info in the log?